arangod/IResearch/IResearchAnalyzerFeature.cpp

```diff
@@ -1644,6 +1717,227 @@ Result IResearchAnalyzerFeature::emplace(
   return {};
 }
 
+Result IResearchAnalyzerFeature::removeAllAnalyzers(TRI_vocbase_t& vocbase) {
+  auto analyzerModificationTrx = createAnalyzerModificationTransaction(server(), vocbase.name());
+  if (analyzerModificationTrx) {
+    auto startRes = analyzerModificationTrx->start();
+    if (startRes.fail()) {
+      return startRes;
+    }
+    auto cleanupResult = cleanupAnalyzersCollection(vocbase.name(),
+                                                    analyzerModificationTrx->buildingRevision());
+    if (cleanupResult.fail()) {
+      return cleanupResult;
+    }
+  }
+  auto* engine = EngineSelectorFeature::ENGINE;
+  TRI_ASSERT(engine && !engine->inRecovery());
+  if (!analyzerModificationTrx) {
+    // no modification transaction. Just truncate
+    auto ctx = transaction::StandaloneContext::Create(vocbase);
+    SingleCollectionTransaction trx(ctx, arangodb::StaticStrings::AnalyzersCollection,
+                                    AccessMode::Type::EXCLUSIVE);
+
+    auto res = trx.begin();
+    if (res.fail()) {
+      return res;
+    }
+
+    OperationOptions options;
+    trx.truncateAsync(arangodb::StaticStrings::AnalyzersCollection, options).get();
+    res = trx.commitAsync().get();
+    if (res.ok()) {
+      invalidate(vocbase);
+    }
+    return res;
+  } else {
+    // as in single analyzer case. First mark for deletion (but all at once)
+    auto stringRev = std::to_string(analyzerModificationTrx->buildingRevision());
+    std::string aql("FOR u IN ");
+    aql.append(arangodb::StaticStrings::AnalyzersCollection).append(" UPDATE u.")
+       .append(arangodb::StaticStrings::KeyString).append(" WITH { ")
+       .append(arangodb::StaticStrings::AnalyzersDeletedRevision)
+       .append(": ").append(stringRev).append("} IN ")
+       .append(arangodb::StaticStrings::AnalyzersCollection);
+
+    auto ctx = transaction::StandaloneContext::Create(vocbase);
+    SingleCollectionTransaction trx(ctx, arangodb::StaticStrings::AnalyzersCollection,
```

Comment:
I am not sure if there are side-effects here when having two `SingleCollectionTransaction` objects in the same scope.
The reason is that the destructor of the base class `transaction::Methods` does some cleanup, and I am not 100% sure this is safe when having two such objects in scope at the same time.
So for safety I would prefer putting each `SingleCollectionTransaction` object in its own nested scope here. That will definitely be safe.